app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/JourneyBaseController.scala

```diff
@@ -118,7 +115,7 @@ abstract class JourneyBaseController[Journey](implicit ec: ExecutionContext)
 
   /** Simple POST action to submit form and update journey. */
   final def simpleActionReadWriteJourney(
-    body: Request[_] => Journey => (Journey, Result)
+    body: RequestWithSessionData[_] => Journey => (Journey, Result)
```

Comment:
I would keep here plain `Request[_]`

Reply:
If we do this then we can't do a BadRequest and reshow the form if there is any form validation failures. We need the RequestWithSessionData to get the implicit messages object needed by the twirl template.

Reply:
I've pushed a fix exposing implicit messages in the scope